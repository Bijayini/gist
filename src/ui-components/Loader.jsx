import React, { Fragment } from 'react';

import theme, { zIndex } from '../lib/theme';

const Loader = () => {
  return (
    <Fragment>
      <div className="loader-overlay">
        <div className="loader-container">
          <div className="loader" />
        </div>
      </div>
      <style jsx="true">
        {`
          .loader-overlay {
            position: fixed;
            left: 0;
            right: 0;
            top: 0;
            bottom: 0;
            background-color: ${theme.loaderOverlay};
            z-index: ${zIndex.loaderOverlay};
            display: flex;
            align-items: center;
            justify-content: center;
          }
          .loader-container {
            width: 200px;
            height: 200px;
            border-radius: 8px;
            box-shadow: 0 2px 6px 0 ${theme.loaderBoxShadow};
            background-color: ${theme.white};
            display: flex;
            align-items: center;
            justify-content: center;
          }
          .loader {
            border: 16px solid ${theme.grey1};
            border-top: 16px solid ${theme.primaryColor};
            border-radius: 50%;
            width: 120px;
            height: 120px;
            animation: spin 1.5s linear infinite;
          }

          @keyframes spin {
            0% {
              transform: rotate(0deg);
            }
            100% {
              transform: rotate(360deg);
            }
          }
        `}
      </style>
    </Fragment>
  );
};

export default Loader;
