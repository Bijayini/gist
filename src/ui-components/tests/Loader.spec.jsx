import React from 'react';
import { shallow } from 'enzyme';

import Loader from '../Loader';

describe('<Loader />', () => {
  let component;

  beforeEach(() => {
    component = shallow(<Loader />);
  });

  it('should match with snapshot', () => {
    expect(component.debug()).toMatchSnapshot();
  });
});
