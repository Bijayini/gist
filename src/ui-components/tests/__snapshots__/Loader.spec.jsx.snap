// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should match with snapshot 1`] = `
"<Fragment>
  <div className=\\"jsx-4235491210 loader-overlay\\">
    <div className=\\"jsx-4235491210 loader-container\\">
      <div className=\\"jsx-4235491210 loader\\" />
    </div>
  </div>
  <JSXStyle id=\\"3461739407\\" dynamic={{...}}>
    .loader-overlay.__jsx-style-dynamic-selector{position:fixed;left:0;right:0;top:0;bottom:0;background-color:rgba(50,50,50,0.95);z-index:999;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}.loader-container.__jsx-style-dynamic-selector{width:200px;height:200px;border-radius:8px;box-shadow:0 2px 6px 0 rgba(0, 0, 0, 0.5);background-color:#ffffff;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}.loader.__jsx-style-dynamic-selector{border:16px solid #4a4a4a;border-top:16px solid #cc1c5d;border-radius:50%;width:120px;height:120px;-webkit-animation:spin-__jsx-style-dynamic-selector 1.5s linear infinite;animation:spin-__jsx-style-dynamic-selector 1.5s linear infinite;}@-webkit-keyframes spin-__jsx-style-dynamic-selector{0%{-webkit-transform:rotate(0deg);-ms-transform:rotate(0deg);transform:rotate(0deg);}100%{-webkit-transform:rotate(360deg);-ms-transform:rotate(360deg);transform:rotate(360deg);}}@keyframes spin-__jsx-style-dynamic-selector{0%{-webkit-transform:rotate(0deg);-ms-transform:rotate(0deg);transform:rotate(0deg);}100%{-webkit-transform:rotate(360deg);-ms-transform:rotate(360deg);transform:rotate(360deg);}}
  </JSXStyle>
</Fragment>"
`;
